src/bsoncxx/test/CMakeLists.txt

```diff
@@ -93,6 +96,25 @@ if(ENABLE_MACRO_GUARD_TESTS)
             DEFINE_NO_DEPRECATED
             BSONCXX_NO_DEPRECATED
             BSONCXX_UNREACHABLE # prelude.hpp
+            _bsoncxxDisableWarningImpl_for_GCC
+            _bsoncxxDisableWarningImpl_for_GNU
+            _bsoncxxDisableWarningImpl_for_MSVC
+            _bsoncxxDisableWarningImpl_for_Clang
+            bsoncxx_push_warnings
+            bsoncxx_pop_warnings
+            bsoncxx_disable_warning
+            bsoncxx_concat
+            bsoncxx_concat_impl
+            bsoncxx_force_semicolon
+            bsoncxx_pragma
+            _bsoncxxPragma
+            bsoncxx_stringify
+            bsoncxx_stringify_impl
+            bsoncxx_if_msvc
+            bsoncxx_if_gcc
+            bsoncxx_if_clang
+            bsoncxx_if_gnu
+            bsoncxx_returns
```

Comment:
Unless there is compelling reason otherwise, recommend using a consistent `UPPER_CASE` + `BSONCXX_`-prefix naming scheme for macros used in public headers.

Reply:
Note: if we want to better distinguish public macros from private macros via their names, rather than solely by documentation, recommend using a `BSONCXX_PRIVATE_` prefix rather than leading underscores (we already have `BSONCXX_PRIVATE` to declare symbol visibility for non-public-API symbols).

Reply:
I don't think there's a need to qualify private macros since we already `push_macro` and `pop_macro` them all before/after every source file, so there is no risk of them leaking out.

The same might be said about using `UPPER_CASE` for all macros, since we aren't going to collide with any names. I'm not _against_ `UPPER_CASE` macros, I just default to `lower_snake_case` for most things.

Reply:
pain. Macros updated.